src/main/java/com/neatresults/mgnltweaks/json/JsonBuilder.java

```diff
@@ -431,6 +438,59 @@ public Object remove(final Object key) {
             return obj;
         }
 
+		@Override
+		public Object get(Object key) {
+			Object superResult = super.get(key);
+			if (!(superResult instanceof ContentMap))
+				return superResult;
+
+			Node node = ((ContentMap) superResult).getJCRNode();
+			if (isArrayParent(node))
+                return childrenAsContentMapList(node);
+
+            return superResult;
+		}
+
+        private boolean isArrayParent(Node candidate) {
+            for (String key : config.childrenArrayCandidates.keySet()) {
+                final String value = config.childrenArrayCandidates.get(key);
+                final Pattern valuePattern = Pattern.compile(value);
+                final Pattern keyPattern = Pattern.compile(key);
+
+                try {
+                    if (!asPropertyStream(candidate.getProperties()).filter(
+							p -> {
+								try {
+									return keyPattern.matcher(p.getName()).matches()
+											&& valuePattern.matcher(p.getString()).matches();
+								} catch (RepositoryException e) {
+									throw new RuntimeException(e);
+								}
+							}).collect(Collectors.toList()).isEmpty())
+						return true;
+                } catch (RepositoryException e) {
+                    throw new RuntimeException(e);
```

Comment:
Hmm, RepositoryException is quite general, but from what I understand, it's likely an indicator of a messed up database. Silently swallowing it and returning a fallback value would be rather irresponsible in this case, as it might become hard to figure out why data is suddenly missing.

Also, error messages would only be printed in edit mode and never shown to end users if I'm not mistaken, right?

But I agree that potentially messing up syntax is a bad idea, so a log message is probably a good middle ground.